src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -75,12 +76,11 @@ private[health] class HealthCheckActor(
   def purgeStatusOfDoneInstances(instances: Seq[Instance]): Unit = {
     logger.debug(s"Purging health status of inactive instances for app ${app.id} version ${app.version} and healthCheck ${healthCheck}")
 
-    val inactiveInstanceIds: Set[Instance.Id] = instances.filterNot(_.isActive).map(_.instanceId)(collection.breakOut)
-    inactiveInstanceIds.foreach { inactiveId =>
-      healthByInstanceId.remove(inactiveId)
-      // Remove inactive (definitively killed) instance from killingInFlight list
-      killingInFlight = killingInFlight - inactiveId
-    }
+    val activeTaskIds: Set[Task.Id] = instances.map(_.appTask).filter(_.isActive).map(_.taskId)(collection.breakOut)
+    healthByTaskId.retain((taskId, health) => activeTaskIds(taskId))
+    // FIXME: I discovered this is unsafe since killingInFlight might be used in 2 concurrent threads (see preStart method above)
+    killingInFlight &= activeTaskIds
+    logger.info(s"[anti-snowball] currently ${killingInFlight.size} instances killingInFlight")
```

Comment:
we don't know for which app this log is so it is effectively useless at info level

Reply:
will fix this by adding the context (app + version), but I'll keep the log at info level (too useful for now)